bob/src/hw_metrics_collector.rs

```diff
@@ -148,6 +151,36 @@ impl DescrCounter {
         //  |  10.000   |      0.006     |
         //  with payload
         //  |  10.000   |      0.018     |
+        if self.lsof_flag {
```

Comment:
I think you should extract it to separate function, `count_descriptors_by_lsof(...) -> Option<u64>`

Reply:
And probably use `if !self.lsof_flag { return None }` in this function to reduce indent levels